src/sentry/api/endpoints/project_details.py

```diff
@@ -647,17 +651,78 @@ def put(self, request: Request, project) -> Response:
         if "dynamicSampling" in result:
             raw_dynamic_sampling = result["dynamicSampling"]
             fixed_rules = self._fix_rule_ids(project, raw_dynamic_sampling)
+
             if project.update_option("sentry:dynamic_sampling", fixed_rules):
+
                 changed_proj_settings["sentry:dynamic_sampling"] = result["dynamicSampling"]
 
+                if old_data.get("dynamicSampling") is not None:
+                    old_rules = old_data["dynamicSampling"].get("rules") or []
+                    rules = result["dynamicSampling"].get("rules") or []
+
+                    if len(rules) > len(old_rules):
+                        self.create_audit_entry(
+                            request=request,
+                            organization=project.organization,
+                            target_object=project.id,
+                            event=audit_log.get_event_id("SAMPLING_RULE_ADD"),
+                            data=project.get_audit_log_data(),
+                        )
+                        data = serialize(project, request.user, DetailedProjectSerializer())
+                        return Response(data)
+
+                    if len(rules) < len(old_rules):
```

Comment:
This endpoint method is pretty massive. This logic looks like it could be moved out into a separate method that returns the audit entries that need to be created.

Reply:
thought about it too but since I am not very familiar with our backend code, I decided to wait for feedback... thanks, Mark! 🥔